File: solutions/1019-squares-of-a-sorted-array/solution.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ans=[0]*len(nums)
        st=0
        ed=len(nums)-1
        pos=len(nums)-1
        while st<=ed:
            if(abs(nums[st])>abs(nums[ed])):
                ans[pos]=nums[st]*nums[st]
                st+=1
            else:
                ans[pos]=nums[ed]*nums[ed]
                ed-=1
            pos-=1
        return ans
        

File: solutions/1019-squares-of-a-sorted-array/solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& arr) {
        int n=arr.size();
        vector<int> res(n);
        int s=0,e=n-1,pos=n-1;
        while(s<=e){
            if(abs(arr[s])>abs(arr[e])){
                res[pos]=arr[s]*arr[s];
                s++;
            }
            else{
                res[pos]=arr[e]*arr[e];
                e--;
            }
            pos--;
        }
        return res;
    }
};
